test/test_dictionary_encoded_array.cpp

```diff
@@ -58,6 +59,11 @@ namespace sparrow
         return ar;
     }
 
+    nullable<std::string_view> get_dict_value(layout_type_cref r)
```

Comment:
Not here, nullable acts as a reference proxy. When the `value_type` of an array is `nullable<T>`, the `const_reference` is usually `nullable<const T&>` (there are some exceptions like `std::string` or `null_type`). For Arays returning a variant, the const_reference is a variant of nullable (objects, because variant cannot hold references) holding references.